tools/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -29,21 +29,85 @@ or logos must follow Microsoft's [Trademark & Brand Guidelines](https://www.micr
 Use of Microsoft trademarks or logos in modified versions of this project must not cause confusion or imply Microsoft sponsorship.
 `;
 
-function packageShouldBePrivate(name: string): boolean {
-    // allow test packages to be packaged
-    if (name.startsWith("@fluid-internal/test-")) {
-        return false;
-    }
+// Some of our package scopes definitely should publish, and others should never publish.  If they should never
+// publish, we want to add the "private": true flag to their package.json to prevent publishing, and conversely if
+// they should always publish we should ensure the "private": true flag is not present.  There is then a third
+// category of packages which may elect whether or not to publish -- we leave the choice up to those individual
+// packages whether to set the flag.
+
+/**
+ * Whether the package is known to be a publicly published package for general use.
+ */
+function packageMustPublishToNPM(name: string): boolean {
+    return (
+        name.startsWith("@fluidframework/")
+        || name === "fluid-framework"
+        || name === "tinylicious"
+    );
+}
+
+/**
+ * Whether the package is known to be an internally published package but not to NPM.
+ * Note that packages published to NPM will also be published internally, however.
+ * This should be a minmal set required for legacy compat of internal partners or internal CI requirements.
+ */
+function packageMustPublishToInternalFeedOnly(name: string): boolean {
+    return (
+        // TODO: We may not need to publish test packages to the internal feed, remove these exceptions if possible.
+        name === "@fluid-internal/test-app-insights-logger"
+        || name === "@fluid-internal/test-service-load"
+        // Most examples should be private, but table-document needs to publish internally for legacy compat
+        || name === "@fluid-example/table-document"
+    );
+}
 
+/**
+ * Whether the package has the option to publicly publish if it chooses.
+ * For example, an experimental package may choose to remain unpublished until it's ready for customers to try it out.
+ */
+function packageMayChooseToPublishToNPM(name: string): boolean {
```

Comment:
Minor: I might remove the NPM part of this since it's really either.

Reply:
Here I chose to be explicit about NPM just because I want that to be the primary decision point -- the internal publishing discrepancies should (IMO) be treated as a weird anomaly that gets the special callout rather than being incorporated into the mainstream publishing scenario.

Reply:
Or put another way, I think that eventually `packageMustPublishToInternalFeedOnly()` should disappear.